appcues/src/main/java/com/appcues/debugger/AppcuesDebuggerManager.kt

```diff
@@ -90,88 +85,84 @@ internal class AppcuesDebuggerManager(
+
+                    layoutParams = LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT).apply {
+                        // adds margin top and bottom according to visible status and navigation bar
+                        ViewCompat.getRootWindowInsets(parentView)?.getInsets(WindowInsetsCompat.Type.systemBars())?.let { insets ->
+                            setMargins(insets.left, insets.top, insets.right, insets.bottom)
                         }
+                    }
 
-                        setContent {
-                            CompositionLocalProvider(
-                                // Debugger is in English and always LTR regardless of app settings
-                                LocalLayoutDirection provides LayoutDirection.Ltr
-                            ) {
-                                MaterialTheme {
-                                    DebuggerComposition(debuggerViewModel) {
-                                        stop()
-                                        it.removeView(this)
-                                    }
+                    setContent {
+                        CompositionLocalProvider(
+                            // Debugger is in English and always LTR regardless of app settings
+                            LocalLayoutDirection provides LayoutDirection.Ltr
+                        ) {
+                            MaterialTheme {
+                                DebuggerComposition(debuggerViewModel) {
+                                    stop()
+                                    parentView.removeView(this)
                                 }
                             }
                         }
                     }
-                )
-            }
+                }
+            )
         }
     }
 
     private fun removeDebuggerView() {
         // does nothing if currentActivity is not initialized
         if (this::currentActivity.isInitialized.not()) return
 
-        getParentView(currentActivity).also {
-            it.findViewById<ComposeView?>(R.id.appcues_debugger_view)?.run {
-                it.removeView(this)
-            }
+        val parentView = currentActivity.getParentView()
+        val debuggerView = parentView.findViewById<ComposeView?>(R.id.appcues_debugger_view)
+        if (debuggerView != null) {
+            parentView.removeView(debuggerView)
         }
     }
 
-    private fun getParentView(activity: Activity): ViewGroup {
-        // if there is any difference in API levels we can handle it here
-        return activity.window.decorView.rootView as ViewGroup
-    }
-
-    private fun setDebuggerBackPressCallback(activity: Activity) {
-        // add onBackPressedDispatcher to handle internally native android back press when debugger is expanded
-        if (activity is ComponentActivity) {
+    private fun setOnBackPressDispatcher(view: ViewGroup) {
+        view.findViewTreeOnBackPressedDispatcherOwner()?.run {
```

Comment:
instead of getting the BackPressDispatcher from ComponentActivity we are finding it from the view. still need to investigate if there is a way to handle backPress when its under our control (AppcuesViewTreeOwner)